quisp/modules/ConnectionManager.cc

```diff
@@ -742,62 +726,67 @@ void ConnectionManager::intermediate_reject_req_handler(RejectConnectionSetupReq
+  int rule_index = 0;
+  std::vector<int> partners = {conf.left_partner, conf.right_partner};
+  RuleSet *EntanglementSwapping = new RuleSet(RuleSet_id, owner, partners);
+  Rule *SwappingRule = new Rule(RuleSet_id, rule_index);
+  EntanglementSwapping->setRule_ptr(SwappingRule);
+  Condition *Swap_condition = new Condition();
+  Clause *resource_clause_left = new EnoughResourceClauseLeft(conf.left_partner, conf.lres);
+  Clause *resource_clause_right = new EnoughResourceClauseRight(conf.right_partner, conf.rres);
+  Swap_condition->addClause(resource_clause_left);
+  Swap_condition->addClause(resource_clause_right);
+  SwappingRule->setCondition(Swap_condition);
+  quisp::rules::Action *swap_action = new SwappingAction(RuleSet_id, rule_index, conf.left_partner, conf.lqnic_type, conf.lqnic_index, conf.lqnic_address, conf.lres,
+                                                         conf.right_partner, conf.rqnic_type, conf.rqnic_index, conf.rqnic_address, conf.rres, conf.self_left_qnic_index,
+                                                         conf.self_left_qnic_type, conf.self_right_qnic_index, conf.self_right_qnic_type);
+  SwappingRule->setAction(swap_action);
+  EntanglementSwapping->addRule(SwappingRule);
+  rule_index++;
+
+  return EntanglementSwapping;
 }
 
-RuleSet* ConnectionManager::generateRuleSet_Tomography(unsigned long RuleSet_id, int owner, int partner, int num_of_measure, QNIC_type qnic_type, int qnic_index, int num_resources){
+RuleSet *ConnectionManager::generateRuleSet_Tomography(unsigned long RuleSet_id, int owner, int partner, int num_of_measure, QNIC_type qnic_type, int qnic_index,
+                                                       int num_resources) {
   // tomography ruleset
   int rule_index = 0;
-  RuleSet* tomography = new RuleSet(RuleSet_id, owner, partner);
-  Rule* Random_measure_tomography = new Rule(RuleSet_id, rule_index);
+  RuleSet *tomography = new RuleSet(RuleSet_id, owner, partner);
+  Rule *Random_measure_tomography = new Rule(RuleSet_id, rule_index);
 
-  Condition* measurement_condition = new Condition();//Technically, there is no condition because an available resource is guaranteed whenever the rule is ran.
-  Clause* count_clause = new MeasureCountClause(num_of_measure, partner, qnic_type , qnic_index, 0);//3000 measurements in total. There are 3*3 = 9 patterns of measurements. So each combination must perform 3000/9 measurements.
-  Clause* resource_clause = new EnoughResourceClause(partner, num_resources);
+  Condition *measurement_condition = new Condition();  // Technically, there is no condition because an available resource is guaranteed whenever the rule is ran.
+  Clause *count_clause =
+      new MeasureCountClause(num_of_measure, partner, qnic_type, qnic_index,
+                             0);  // 3000 measurements in total. There are 3*3 = 9 patterns of measurements. So each combination must perform 3000/9 measurements.
```

Comment:
These lines are formatted uglily because the comment is put on the same line.
I think clang-format cannot fix this properly, so I'll fix it manually later.